src/plugins/vis_builder/public/application/utils/use/use_saved_vis_builder_vis.ts

```diff
@@ -61,6 +62,19 @@ export const useSavedVisBuilderVis = (visualizationIdFromUrl: string | undefined
           const { title, state } = getStateFromSavedObject(savedVisBuilderVis);
           chrome.setBreadcrumbs(getEditBreadcrumbs(title, navigateToApp));
           chrome.docTitle.change(title);
+          // sync initial app filters from savedObject to filterManager
+          const filters = savedVisBuilderVis.searchSourceFields.filter;
+          const query =
+            savedVisBuilderVis.searchSourceFields.query || data.query.queryString.getDefaultQuery();
+          const actualFilters = [];
+          if (filters !== undefined) {
+            const result = typeof filters === 'function' ? filters() : filters;
+            if (result !== undefined) {
+              actualFilters.push(...(Array.isArray(result) ? result : [result]));
+            }
+          }
```

Comment:
We probably want to validate each filter before adding it to `actualFilters`. Also, I don't think checking that a filter is just not `undefined` is good validation. At the very least making sure it is not falsy (null, undefined, ...) is warranted: 
```suggestion
          const tempFilters = typeof filters === 'function' ? filters() : filters;
          (Array.isArray(tempFilters) ? tempFilters : [tempFilters]).forEach((filter) => {
            if (filter) actualFilters.push(filter);
          });
```
If there is a better way to make sure a filter is a `Filter`, we should do that.